src/main/java/org/reaktivity/nukleus/tls/internal/TlsNukleus.java

```diff
@@ -138,48 +140,45 @@ private boolean handleRoute(
         return true;
     }
 
-    private void handleRoute(
+    private boolean handleRoute(
         final RouteFW route)
     {
         final TlsRouteExFW routeEx = route.extension().get(tlsRouteExRO::wrap);
         final String store = routeEx.store().asString();
         final long routeId = route.correlationId();
 
-        if (store != null)
+        storesByRouteId.put(routeId, store);
+        StoreInfo storeInfo = contextsByStore.computeIfAbsent(store, s -> initContext(config, store, ++storeIndex));
```

Comment:
If we can make `initContext` a non-static instance method, then it can reference `storeIndex` and `config` in the method body instead of as parameters, and then you can pass `this::initContext` as a pure lambda.

Also, let's rename `initContext` to `newStoreInfo`, so the lambda would be `this::newStoreInfo`.